test/ui/PrintableUriTest.ts

```diff
@@ -0,0 +1,228 @@
+import * as Mock from '../MockEnvironment';
+import { PrintableUri } from '../../src/ui/PrintableUri/PrintableUri';
+import { IQueryResult } from '../../src/rest/QueryResult';
+import { $$ } from '../../src/utils/Dom';
+import { IResultLinkOptions } from '../../src/ui/ResultLink/ResultLinkOptions';
+import { analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+import { FakeResults } from '../Fake';
+
+export function PrintableUriTest() {
+  describe('PrintableUri', function () {
+    let test: Mock.IBasicComponentSetup<PrintableUri>;
+    let fakeResult: IQueryResult;
+
+    beforeEach(() => {
+      fakeResult = initFakeResult();
+      test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
+      spyOn(test.cmp, 'openLink');
+      spyOn(window, 'open');
+    });
+
+    afterEach(function () {
+      test = null;
+      fakeResult = null;
+    });
+
+    it('should have its tabindex value set to 0', () => {
+      expect(test.cmp.element.getAttribute('tabindex')).toBe('0');
+    });
+
+    it('should display the XML correctly if the result has a non-null parents field', () => {
+      fakeResult.raw.parents = '<?xml version="1.0" encoding="utf-16"?><parents><parent name="My Drive" uri="https://drive.google.com/#my-drive" /></parents>';
+      test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
+      expect($$(test.cmp.element).find('span').innerText).toEqual('My Drive');
+    });
+
+    it('should display the XML correctly if the result has a very long parents field', () => {
+      fakeResult.raw.parents = '<?xml version="1.0" encoding="utf-16"?><parents><parent name="Organization" uri="https://na17.salesforce.com/home/home.jsp" /><parent name="Technical_Article__ka" uri="http://www.salesforce.com/org:organization/articletype:Technical_Article" /><parent name="Generator became sentient and refuses to cut power" uri="https://na17.salesforce.com/kA0o00000003Wpk" /><parent name="un autre test" uri="http//:google.ca" /></parents>';
```

Comment:
Just a suggestion : you can use template strings using a backtick (`) to write multi-line strings. It makes the code look a bit better.